Content.Shared/Whitelist/EntityWhitelist.cs

```diff
@@ -1,100 +1,119 @@
+using Content.Shared.Item;
 using Content.Shared.Tag;
 using Robust.Shared.Serialization;
 using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.List;
 
-namespace Content.Shared.Whitelist
+namespace Content.Shared.Whitelist;
+
+/// <summary>
+///     Used to determine whether an entity fits a certain whitelist.
+///     Does not whitelist by prototypes, since that is undesirable; you're better off just adding a tag to all
+///     entity prototypes that need to be whitelisted, and checking for that.
+/// </summary>
+/// <code>
+/// whitelist:
+///   tags:
+///   - Cigarette
+///   - FirelockElectronics
+///   components:
+///   - Buckle
+///   - AsteroidRock
+///   sizes:
+///   - Tiny
+///   - Large
+/// </code>
+[DataDefinition]
+[Serializable, NetSerializable]
+public sealed partial class EntityWhitelist
 {
     /// <summary>
-    ///     Used to determine whether an entity fits a certain whitelist.
-    ///     Does not whitelist by prototypes, since that is undesirable; you're better off just adding a tag to all
-    ///     entity prototypes that need to be whitelisted, and checking for that.
+    ///     Component names that are allowed in the whitelist.
     /// </summary>
-    /// <code>
-    /// whitelist:
-    ///   tags:
-    ///     - Cigarette
-    ///     - FirelockElectronics
-    ///   components:
-    ///     - Buckle
-    ///     - AsteroidRock
-    /// </code>
-    [DataDefinition]
-    [Serializable, NetSerializable]
-    public sealed partial class EntityWhitelist
-    {
-        /// <summary>
-        ///     Component names that are allowed in the whitelist.
-        /// </summary>
-        [DataField("components")] public string[]? Components = null;
-        // TODO yaml validation
+    [DataField] public string[]? Components;
+    // TODO yaml validation
 
-        [NonSerialized]
-        private List<ComponentRegistration>? _registrations = null;
+    /// <summary>
+    ///     Item sizes that are allowed in the whitelist.
+    /// </summary>
+    [DataField("sizes", customTypeSerializer:typeof(PrototypeIdListSerializer<ItemSizePrototype>))]
+    public List<string>? Sizes;
```

Comment:
```suggestion
    [DataField]
    public List<ProtoId<ItemSizePrototype>>? Sizes;
```
modern ss14 techniques